site/app/controllers/student/SubmissionController.php

```diff
@@ -786,13 +786,17 @@ private function ajaxUploadSplitItem() {
         if (!@file_put_contents(FileUtils::joinPaths($version_path, ".submit.timestamp"), $current_time_string_tz."\n")) {
             return $this->uploadResult("Failed to save timestamp file for this submission.", false);
         }
-        $upload_time_string_tz = $timestamp . " " . $this->core->getConfig()->getTimezone()->getName() . "\n";
+
+        $upload_time_string_tz = $timestamp . " " . $this->core->getConfig()->getTimezone()->getName();
        
-        $bulk_upload_data_json = array("submit_timestamp" =>  $current_time_string_tz,
-                                       "upload_timestamp" =>  $upload_time_string_tz,
-                                       "filepath" => $uploaded_file);
+        $bulk_upload_data = array("submit_timestamp" =>  $current_time_string_tz,
+                                  "upload_timestamp" =>  $upload_time_string_tz,
+                                  "filepath" => $uploaded_file);
+        
+        $bulk_upload_data_json = json_encode($bulk_upload_data, JSON_PRETTY_PRINT);
         
-        if (!@file_put_contents(FileUtils::joinPaths($version_path, "bulk_upload_data.json"), serialize($bulk_upload_data_json)."\n")) {
+        if (!@file_put_contents(FileUtils::joinPaths($version_path, "bulk_upload_data.json"), $bulk_upload_data_json."\n")) { 
+
             return $this->uploadResult("Failed to create bulk upload file for this submission.", false);
         }
```

Comment:
Use `FileUtils:;writeJsonFile` for writing PHP arrays to JSON files. There's not a ton of point to adding the `\n` to the end of the file.
```
        $bulk_upload_data = [
            "submit_timestamp" =>  $current_time_string_tz,
            "upload_timestamp" =>  $upload_time_string_tz,
            "filepath" => $uploaded_file
        ];

        if (FileUtils::writeJsonFile(FileUtils::joinPaths($version_path, "bulk_upload_data.json"), $bulk_upload_data)) {
            return $this->uploadResult("Failed to create bulk upload file for this submission.", false);
        }
```